### PR TITLE
Implement OsuModNinja

### PR DESCRIPTION
**Pull request by EVAST9919**
Inspired by FruitNinja

Preview: https://www.youtube.com/watch?v=ts_XkVjbJ-Q

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Mods/OsuModNinja.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    internal class OsuModNinja : Mod, IApplicableToDrawableHitObjects
+    {
+        public override string Name => "Ninja";
+        public override string Acronym => "NJ";
+        public override IconUsage? Icon => FontAwesome.Solid.UserNinja;
+        public override ModType Type => ModType.Fun;
+        public override string Description => "Slice the circles!";
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Mods/OsuModNinja.cs:23]: this name and description seem misleading. when I think of what is the core mechanic in fruit ninja, I think of slicing. this mod doesn't *actually* add slicing; from what I can see it solely has the hit objects emulate the way fruits fly into the playfield. (they don't fly back out of the screen either, just hang in their final position.)

(also not sure of potential legal repercussions of naming this after that game.)

File: osu.Game.Rulesets.Osu/Mods/OsuModNinja.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                case DrawableSliderRepeat _:
+                    return;
+
+                default:
+                    var hitObject = (OsuHitObject)drawable.HitObject;
+
+                    var originalPosition = drawable.Position;
+                    var appearPosition = new Vector2(originalPosition.X, OsuPlayfield.BASE_SIZE.Y);
+
+                    // the - 1 and + 1 prevents the hit objects to appear in the wrong position.
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Mods/OsuModNinja.cs:49]: mind explaining this?

**Comment by bdach** [comment]
a proper implementation mimicking mechanics of fruit ninja could actually be a decent idea for its own ruleset, but I think it wouldn't be suited to be a mod due to implementation complexity.

File: osu.Game.Rulesets.Osu/Mods/OsuModNinja.cs (hunk 2 above)
Comment (EVAST9919) [reply osu.Game.Rulesets.Osu/Mods/OsuModNinja.cs:49]: Same as for Transform mod

**Comment by EVAST9919** [comment]
I just thought it'd be nice to have (more like another variation of Transform mod)

**Comment by EVAST9919** [comment]
> a proper implementation mimicking mechanics of fruit ninja could actually be a decent idea for its own ruleset, but I think it wouldn't be suited to be a mod due to implementation complexity.

sounds good. It's easier to make a new ruleset with this mechanics in mind rather then trying to implement something very complex

**Pull request closed without merging**